src/components/MultipieceModal/MultipieceModal.test.js

```diff
@@ -0,0 +1,240 @@
+    itemDescriptionOfPieces: 'ui-checkin.multipieceModal.item.descriptionOfPieces',
+    itemNumberOfMissingPieces: 'ui-checkin.multipieceModal.item.numberOfMissingPieces',
+    itemDescriptionOfMissingPieces: 'ui-checkin.multipieceModal.item.descriptionOfmissingPieces',
+  };
+  const testOpen = false;
+  const testItem = {
+    title: 'testTitle',
+    barcode: 'testBarcode',
+    materialType: {
+      name: 'testName',
+    },
+  };
+  const testOnClose = jest.fn();
+  const testOnConfirm = jest.fn();
+  const testDefaultProps = {
+    onClose: testOnClose,
+    onConfirm: testOnConfirm,
+    item: testItem,
+    open: testOpen,
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('with default props', () => {
+    beforeEach(() => {
+      render(<MultipieceModal {...testDefaultProps} />);
+    });
+
+    it('should render multipiece "Modal" component', () => {
+      componentPropsCheck(Modal, testIds.multipieceModal, {
+        id: 'multipiece-modal',
+        size: 'small',
+        dismissible: true,
+        label: labelIds.multipieceModalLabel,
+      }, true);
+    });
+
+    it('should render multipiece "ModalFooter" component', () => {
+      componentPropsCheck(ModalFooter, testIds.multipieceModalFooter, {}, true);
+    });
+
+    it('should render multipiece modal confirm "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalConfirmButton, {
+        buttonStyle: 'primary',
+      }, true);
+    });
+
+    it('should render multipiece modal confirm "Button" component label', () => {
+      expect(getById(testIds.multipieceModalConfirmButton).getByText(labelIds.multipieceModalCheckIn)).toBeVisible();
+    });
+
+    it('should call onConfirm callback when click on confirm button', () => {
+      expect(testOnConfirm).not.toBeCalled();
+
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalConfirmButton));
+
+      expect(testOnConfirm).toHaveBeenCalled();
+    });
+
+    it('should render multipiece modal cancel "Button" component', () => {
+      componentPropsCheck(Button, testIds.multipieceModalCancelButton, {}, true);
+    });
+
+    it('should render multipiece modal cancel "Button" component label', () => {
+      expect(getById(testIds.multipieceModalCancelButton).getByText(labelIds.multipieceModalCancel)).toBeVisible();
+    });
+
+    it('should call onClose callback when click on cancel button', () => {
+      expect(testOnClose).not.toBeCalled();
+
+      fireEvent.click(screen.getByTestId(testIds.multipieceModalCancelButton));
+
+      expect(testOnClose).toHaveBeenCalled();
+    });
+
+    it('should render multipiece modal message label', () => {
+      expect(screen.getByText(labelIds.multipieceModalMessage)).toBeVisible();
+    });
+
+    it('should render number of pieces "KeyValue" component', () => {
+      componentPropsCheck(KeyValue, testIds.numberOfPieces, {
+        label: labelIds.itemNumberOfPieces,
+        value: '-',
```

Comment:
Could you add `value` variable?
I see that we have 4 cases when `value` equal `-`

Reply:
Done